misc/genstd/mapping.go

```diff
@@ -231,15 +231,95 @@
 			return err
 		}
 		return nil
+	case *ast.MapType:
+		goe, ok := goe.(*ast.MapType)
+		if !ok {
+			return &mismatch
+		}
+		if err := m.typesEqual(gnoe.Key, goe.Key); err != nil {
```

Comment:
Do these errors need to be returned raw, or can you use `fmt.Errorf` to give them a bit more context?